### PR TITLE
SERENE-741: Asynchronous index ingestion and status update

### PR DESCRIPTION
**Pull request by gaj67**
Perhaps test with the full IMDB data-set and check that the HTTP request returns before ingestion is finished, and that the index status changes from INDEXING to COMPLETE after ingestion has completed.

### CODE REVIEW

**Comment by gaj67** [comment]
Tested by copying full IMDB example to docker/data, and using Postman create with:

{
  "graphs": "file:/data/EPGM_Examples/imdb.json/graphs.json",
  "edges": "file:/data/EPGM_Examples/imdb.json/edges.json",
  "vertices": "file:/data/EPGM_Examples/imdb.json/vertices.json"
}

Note: You will need to delete the CRC files.

File: search/src/main/java/au/com/d2dcrc/ia/search/management/EpgIngestionTask.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            Collection<EpgElasticsearchNode> epgNodes = EpgElasticsearchTransformer.getNodes(epgData);
+
+            // Insert the ElasticSearch nodes into the index.
+            elasticsearchIndexHelper.addToIndex(indexName, epgNodes);
+
+            // Update the meta index status.
+            setIndexStatus(EpgStatus.COMPLETE);
+            logger.debug("Successfully ingested EPG data into index: " + indexName);
+        } catch (RuntimeException e) {
+            logger.debug("Failed to properly ingest EPG data into index: " + indexName, e);
```
Comment (tmoschou) [inline search/src/main/java/au/com/d2dcrc/ia/search/management/EpgIngestionTask.java]: logger.error

**Comment by tmoschou** [comment]
The CRC files I think we could programatically ignore since the filenames start with a period `.` to denote a hidden file - which is easy enough to add as a filter to the tree-walker. And will ease integration with upstream modules. With postman, another solution might be to have
```
{
"graphs": "file:{{data_dir}}/EPGM_Examples/imdb.json/graphs.json",
"edges": "file:{{data_dir}}/EPGM_Examples/imdb.json/edges.json",
"vertices": "file:{{data_dir}}/EPGM_Examples/imdb.json/vertices.json"
}
```
and have `data_dir` defined different in a `docker` or `standalone` environment. Which will avoid needing to create a symlink as root (and hence highly likely the link will be forgotten to be deleted when no longer needed).

File: search/src/main/java/au/com/d2dcrc/ia/search/management/EpgIngestionTask.java (hunk 1 above)
Comment (gaj67) [reply search/src/main/java/au/com/d2dcrc/ia/search/management/EpgIngestionTask.java]: Done

**Pull request merged**